### PR TITLE
Nested Struct encoding example

### LINKED ISSUE #1095: feat(`docs`): Add more complex/nested encoding examples
Right now our encoding examples are pretty basic. We could do better by showing some complex examples, such as structs, nested structs, so on and so forth. This would be ideally useful for any deployments tools, `cast call`, and `cast send`.

### PR DESCRIPTION
**Pull request by kamuik16**
Closes #1095

### CODE REVIEW

**Comment by kamuik16** [comment]
Ignore the cast balance commit. It already got merged in another PR https://github.com/foundry-rs/book/pull/1128.

File: src/misc/struct-encoding.md [hunk 1]
```diff
@@ -61,3 +61,72 @@ The ABI of the `f` function in this contract is:
...
+            ]
+        }
+    ],
+    "name": "f",
+    "outputs": [],
+    "stateMutability": "pure",
+    "type": "function"
+}
+```
+which reads: The function `f` takes 1 input of type tuple with three components: a string, a uint256, and another tuple representing the nested struct with components addr of type address and amount of type uint256.
```
Comment (onbjerg) [inline src/misc/struct-encoding.md:132]: can we add an example of what this looks like when invoked? otherwise lgtm

Comment (onbjerg) [reply src/misc/struct-encoding.md:132]: like how you would pass it to e.g. constructor args for `forge create`, or for abi encoding with `cast abi-encode`

Comment (kamuikatsurgi) [reply src/misc/struct-encoding.md:132]: > like how you would pass it to e.g. constructor args for `forge create`, or for abi encoding with `cast abi-encode`

working on it.

Comment (kamuikatsurgi) [reply src/misc/struct-encoding.md]: > like how you would pass it to e.g. constructor args for `forge create`, or for abi encoding with `cast abi-encode`

added examples for both `forge create` and `cast abi-encode`.

Comment (kamuik16) [reply src/misc/struct-encoding.md]: > > like how you would pass it to e.g. constructor args for `forge create`, or for abi encoding with `cast abi-encode`
> 
> added examples for both `forge create` and `cast abi-encode`.

accidentally replied from another acc.

Comment (kamuik16) [reply src/misc/struct-encoding.md]: > like how you would pass it to e.g. constructor args for `forge create`, or for abi encoding with `cast abi-encode`

added examples for both `forge create` and cast `abi-encode`.

**Pull request merged**